src/utils.js

```diff
@@ -58,6 +58,33 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  variables.forEach((variable) => {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+    }
+  });
+
+  if (typeof scopeVar === 'undefined' ||
+      typeof scopeVar.defs === 'undefined' ||
+      typeof scopeVar.defs[0] === 'undefined' ||
+      typeof scopeVar.defs[0].node.init === 'undefined' ||
```

Comment:
I would invert the checks, maybe something like this:

```javascript
if (scopeVar && scopeVar.defs && scopeVar.defs[0] && 
    scopeVar.defs[0].node && scopeVar.defs[0].node.init) {
  return scopeVar.defs[0].node.init.name; 
} else {
  return node.name;
}
```

Of course this assumes non-falsey values.